### PR TITLE
Ignore ping commands in MongoDB

### PR DESCRIPTION
**Pull request by kevingosse**
This should fix this kind of issue: https://github.com/DataDog/dd-trace-dotnet/issues/2168

Also, rewrite integration tests to use fluent assertions.

### CODE REVIEW

File: tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs [hunk 1]
```diff
@@ -101,6 +101,12 @@ internal static Scope CreateScope(object wireProtocol, object connection)
                     {
                         firstElement.TryGetPropertyValue("Name", out operationName);
 
+                        if (operationName == "isMaster" || operationName == "hello")
+                        {
+                            // This is the command used by MongoDb for ping operations
+                            return null;
+                        }
+
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs:109]: Given this is the only "bail out" point in this method, and that these ping operations are called a lot, is it worth rearranging this method a bit, so we bail out earlier?

**Review by andrewlock — APPROVED** [review]

**Review by pierotibou — APPROVED** [review]

File: tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs (hunk 1 above)
Comment (kevingosse) [reply tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs]: Done in https://github.com/DataDog/dd-trace-dotnet/pull/2216/commits/4314fe9fa66e266277584869b618dbf783faad9e

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2216 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f965b92870ad31742fa9d9329650d4d38896f914) will **not change** line coverage
:heavy_check_mark: Merging #2216 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f965b92870ad31742fa9d9329650d4d38896f914) will **not change** branch coverage
:no_entry: Merging #2216 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/f965b92870ad31742fa9d9329650d4d38896f914) will will **increase** complexity by `4`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f965b92870ad31742fa9d9329650d4d38896f914) | #2216       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `8967` / `12180` | `8957` / `12182` |          |
| Lines %   | `74%`      | `74%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4095` / `5911` | `4088` / `5915` |          |
| Branches %| `69%`      | `69%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7095`      | `7099`        |  `4`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84743&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84743/artifacts?artifactName=Code%20Coverage%20Report_84743&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2216](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84788&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84788/artifacts?artifactName=Code%20Coverage%20Report_84788&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/f965b92870ad31742fa9d9329650d4d38896f914) | #2216       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `74%`      | `74%`       |  `0%` :heavy_check_mark:  |
| Branches %| `69%`      | `69%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7095`      | `7099`       |  `4` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Couchbase.IIOServiceExecuteIntegrationBis](https://github.com/DataDog/dd-trace-dotnet/tree/41f4016c1e0ff294093c577d95f567058a3c1cfd/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Couchbase/IIOServiceExecuteIntegrationBis.cs) | `-100%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.MsTestV2.TestMethodRunnerExecuteIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/41f4016c1e0ff294093c577d95f567058a3c1cfd/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs) | `-33%` :no_entry: | `-19%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.NUnit.NUnitCompositeWorkItemSkipChildrenIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/41f4016c1e0ff294093c577d95f567058a3c1cfd/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitCompositeWorkItemSkipChildrenIntegration.cs) | `-8%` :no_entry: | `-8%` :no_entry:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84743&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84743/artifacts?artifactName=Code%20Coverage%20Report_84743&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2216](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=84788&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/84788/artifacts?artifactName=Code%20Coverage%20Report_84788&api-version=6.0&%24format=zip)

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #2216 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6): 
* 2 benchmarks are **faster**, with geometric mean 1.210
* 2 benchmarks are **slower**, with geometric mean 1.185
* 2 benchmarks have **fewer** allocations
* 1 benchmarks have **more** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Fewer allocations :tada:



#### Fewer allocations :tada: in #2216

|  Benchmark         | Base Allocated | Diff Allocated | Change   | Change % |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.AgentWriterBenchmark.WriteAndFlushEnrichedTraces&#8209;netcoreapp3.1 | 2.57 KB | 2.57 KB | -1 B | -0.04%
| Benchmarks.Trace.AgentWriterBenchmark.WriteAndFlushEnrichedTraces&#8209;net472 | 3.17 KB | 3.16 KB | -3 B | -0.09%

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`WriteAndFlushEnrichedTraces`|net472|998μs|2.28μs|8.54μs|0| 0|0|3.17&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|785μs|1.7μs|6.57μs|0| 0|0|2.57&nbsp;KB|
|#2216|`WriteAndFlushEnrichedTraces`|net472|978μs|2.66μs|9.96μs|0| 0|0|3.16&nbsp;KB|
|#2216|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|754μs|3.84μs|18μs|0| 0|0|2.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`SendRequest`|netcoreapp3.1|304μs|1.61μs|7.87μs|0.147| 0|0|19.59&nbsp;KB|
|#2216|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#2216|`SendRequest`|netcoreapp3.1|299μs|1.45μs|5.79μs|0.148| 0|0|19.59&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`ExecuteNonQuery`|net472|1.77μs|8.66ns|35.7ns|0.0904| 0.000895|0|570&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`ExecuteNonQuery`|netcoreapp3.1|1.61μs|9.33ns|84.5ns|0.00858| 0|0|608&nbsp;B|
|#2216|`ExecuteNonQuery`|net472|1.84μs|9.31ns|51.8ns|0.0903| 0.000852|0|570&nbsp;B|
|#2216|`ExecuteNonQuery`|netcoreapp3.1|1.49μs|3.53ns|13.7ns|0.0082| 0|0|608&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #2216

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.ElasticsearchBenchmark.CallElasticsearch&#8209;net472 | 1.160 | 2,647.68 | 3,071.02 | 



#### Faster :tada: in #2216

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.ElasticsearchBenchmark.CallElasticsearchAsync&#8209;net472 | 1.170 | 3,307.67 | 2,826.60 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`CallElasticsearch`|net472|2.65μs|13ns|55.2ns|0.121| 0|0|778&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`CallElasticsearch`|netcoreapp3.1|1.68μs|6.55ns|24.5ns|0.0103| 0|0|768&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`CallElasticsearchAsync`|net472|3.31μs|17.5ns|90.9ns|0.142| 0|0|915&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`CallElasticsearchAsync`|netcoreapp3.1|1.84μs|8.37ns|31.3ns|0.0126| 0|0|888&nbsp;B|
|#2216|`CallElasticsearch`|net472|3.08μs|10.8ns|40.4ns|0.12| 0|0|778&nbsp;B|
|#2216|`CallElasticsearch`|netcoreapp3.1|1.64μs|6.79ns|29.6ns|0.0104| 0|0|768&nbsp;B|
|#2216|`CallElasticsearchAsync`|net472|2.82μs|7.62ns|29.5ns|0.142| 0|0|915&nbsp;B|
|#2216|`CallElasticsearchAsync`|netcoreapp3.1|1.9μs|9.64ns|43.1ns|0.0124| 0|0|888&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`ExecuteAsync`|net472|3.19μs|15.7ns|64.7ns|0.162| 0|0|1.04&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`ExecuteAsync`|netcoreapp3.1|1.97μs|8.13ns|29.3ns|0.0145| 0|0|1.01&nbsp;KB|
|#2216|`ExecuteAsync`|net472|3.34μs|13.9ns|77.6ns|0.161| 0|0|1.04&nbsp;KB|
|#2216|`ExecuteAsync`|netcoreapp3.1|2.05μs|3.17ns|11.4ns|0.0143| 0|0|1.01&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`SendAsync`|net472|6.87μs|37.5ns|209ns|0.351| 0|0|2.24&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`SendAsync`|netcoreapp3.1|4.57μs|18.1ns|70ns|0.0295| 0|0|2.14&nbsp;KB|
|#2216|`SendAsync`|net472|6.77μs|21.7ns|84ns|0.349| 0|0|2.24&nbsp;KB|
|#2216|`SendAsync`|netcoreapp3.1|4.56μs|15.6ns|60.5ns|0.0296| 0|0|2.14&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|net472|4.15μs|18.9ns|70.8ns|0.22| 0|0|1.41&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|netcoreapp3.1|3.31μs|15.3ns|59.2ns|0.0213| 0|0|1.49&nbsp;KB|
|#2216|`EnrichedLog`|net472|3.79μs|19.8ns|98.8ns|0.221| 0|0|1.41&nbsp;KB|
|#2216|`EnrichedLog`|netcoreapp3.1|3.35μs|16.5ns|79.3ns|0.0215| 0|0|1.49&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: More allocations :warning:



#### More allocations :warning: in #2216

|  Benchmark         | Base Allocated | Diff Allocated | Change   | Change % |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.Log4netBenchmark.EnrichedLog&#8209;netcoreapp3.1 | 4.17 KB | 4.17 KB | 1 B | 0.02%

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|net472|179μs|464ns|1.74μs|0.529| 0.176|0|4.29&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|netcoreapp3.1|160μs|734ns|3.44μs|0| 0|0|4.17&nbsp;KB|
|#2216|`EnrichedLog`|net472|180μs|712ns|2.76μs|0.534| 0.178|0|4.29&nbsp;KB|
|#2216|`EnrichedLog`|netcoreapp3.1|153μs|303ns|1.17μs|0| 0|0|4.17&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|net472|8.52μs|30.3ns|117ns|0.499| 0|0|3.19&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|netcoreapp3.1|7.1μs|38.3ns|213ns|0.0479| 0|0|3.56&nbsp;KB|
|#2216|`EnrichedLog`|net472|8.29μs|24.6ns|95.3ns|0.499| 0|0|3.19&nbsp;KB|
|#2216|`EnrichedLog`|netcoreapp3.1|6.7μs|19.3ns|74.6ns|0.0504| 0|0|3.56&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`SendReceive`|net472|2.59μs|14ns|81.8ns|0.154| 0|0|987&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`SendReceive`|netcoreapp3.1|2.09μs|7.27ns|34.1ns|0.0137| 0|0|984&nbsp;B|
|#2216|`SendReceive`|net472|2.89μs|12.9ns|50.1ns|0.154| 0|0|987&nbsp;B|
|#2216|`SendReceive`|netcoreapp3.1|2.17μs|7.25ns|28.1ns|0.0133| 0|0|984&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #2216

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SerilogBenchmark.EnrichedLog&#8209;net472 | 1.211 | 6,886.05 | 8,340.01 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|net472|6.89μs|33.4ns|138ns|0.285| 0|0|1.83&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`EnrichedLog`|netcoreapp3.1|5.77μs|18.5ns|71.5ns|0.02| 0|0|1.45&nbsp;KB|
|#2216|`EnrichedLog`|net472|8.34μs|34.6ns|134ns|0.283| 0|0|1.83&nbsp;KB|
|#2216|`EnrichedLog`|netcoreapp3.1|5.71μs|28.2ns|159ns|0.0198| 0|0|1.45&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #2216

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishScope&#8209;net472 | 1.252 | 1,483.56 | 1,185.16 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`StartFinishSpan`|net472|1.01μs|4.98ns|22.3ns|0.0677| 0|0|433&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`StartFinishSpan`|netcoreapp3.1|943ns|3.63ns|13.1ns|0.00607| 0|0|432&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`StartFinishScope`|net472|1.49μs|7.6ns|35.7ns|0.08| 0|0|514&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/817d84eb0fc35256fa18865b9b0bfff705cee6c6)|`StartFinishScope`|netcoreapp3.1|1.11μs|5.12ns|19.8ns|0.00787| 0|0|552&nbsp;B|
|#2216|`StartFinishSpan`|net472|984ns|4.91ns|22ns|0.0681| 0|0|433&nbsp;B|
|#2216|`StartFinishSpan`|netcoreapp3.1|948ns|2.43ns|9.4ns|0.00603| 0|0|432&nbsp;B|
|#2216|`StartFinishScope`|net472|1.18μs|4.07ns|15.8ns|0.0804| 0|0|514&nbsp;B|
|#2216|`StartFinishScope`|netcoreapp3.1|1.07μs|5.09ns|22.2ns|0.00743| 0|0|552&nbsp;B|
</details>

**Pull request merged**